android/src/main/java/com/asterinet/react/tcpsocket/TcpSocketServer.java

```diff
@@ -103,4 +107,8 @@ public void close() {
             mReceiverListener.onClose(getId(), e.getMessage());
         }
     }
+
+    public int getListeningPort() {
+        return listeningPort;
```

Comment:
There is no need for a `listeningPort` attribute. `serverSocket.getLocalPort()` returns the port number to which this socket is listening or -1 if the socket is not bound yet. You could just use `serverSocket.getLocalPort()` here.